check_diff/tests/git.rs

```diff
@@ -0,0 +1,34 @@
+use check_diff::clone_git_repo;
+
+use std::fs;
+use tempfile::Builder;
+use tracing::error;
+
+#[test]
+fn clone_repo_test() {
+    let dir = Builder::new().tempdir_in("");
+    match dir {
+        Ok(d) => {
+            let sample_repo = "https://github.com/rust-lang/rustfmt.git";
+            let dest_path = d.path();
+            let result = clone_git_repo(sample_repo, dest_path);
+            assert_eq!(result.is_ok(), true);
+            let directory = fs::read_dir(dest_path);
+            // check whether we can read this directory
+            assert_eq!(directory.is_err(), false);
+            // check that the directory is non-empty
+            assert_eq!(directory.iter().next().is_none(), false);
```

Comment:
Since we've cloned a git repo into this repository we might want to double check that a `.git` directory exists with the `tempdir`. That might be a more direct assertion. Checking that we're able to reach the directly could be useful, but my assumption would be that if we're able to write to the tempdir then we'd be able to read from it as well.